### PR TITLE
build: clean up java dep structure

### PR DESCRIPTION
**Pull request by goaway**

### CODE REVIEW

File: library/java/io/envoyproxy/envoymobile/engine/BUILD [hunk 1]
```diff
@@ -0,0 +1,18 @@
+licenses(["notice"])  # Apache 2
+
+# These source files must be re-exported to the kotlin custom library rule to ensure their
+# inclusion.
+filegroup(
```
Comment (buildbreaker) [inline library/java/io/envoyproxy/envoymobile/engine/BUILD]: Ah! This is how to achieve this

**Comment by buildbreaker** [comment]
I didn't realize we could use `filegroup` to expose a file to an external `BUILD` module.

Could we also move the current `library/BUILD` into the `library/kotlin` directory to further consolidate the build rules to the platform?

**Comment by goaway** [comment]
@buildbreaker yep was going to try that next

**Comment by buildbreaker** [comment]
Talked offline, we'll move the kotlin rules in another PR

**Comment by goaway** [comment]
thanks @buildbreaker!

File: library/java/io/envoyproxy/envoymobile/engine/BUILD (hunk 1 above)
Comment (goaway) [reply library/java/io/envoyproxy/envoymobile/engine/BUILD]: Yep, it also means if we ever add more java srcs, we don't have to worry about missing them in the kotlin rule. :)

**Pull request merged**